target_chains/sui/scripts/pyth/create_all_price_feeds.ts

```diff
@@ -1,118 +1,63 @@
 import * as _ from "lodash";
-import {
-  RawSigner,
-  SUI_CLOCK_OBJECT_ID,
-  TransactionBlock,
-  JsonRpcProvider,
-  Ed25519Keypair,
-  Connection,
-} from "@mysten/sui.js";
+import { Ed25519Keypair } from "@mysten/sui.js";
 
-dotenv.config({ path: "~/.env" });
+import { DefaultStore, SuiContract } from "@pythnetwork/pyth-contract-manager";
 
-import { REGISTRY, NETWORK } from "../registry";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
 
-// ================== Network dependent settings ==================
-let network = NETWORK.MAINNET; // <= NOTE: Update this when changing network
-const walletPrivateKey = process.env.SUI_MAINNET; // <= NOTE: Update this when changing network
-const price_feed_id_url = "https://xc-mainnet.pyth.network/api/price_feed_ids"; // <= NOTE: Update this when changing network
-const connection = new PriceServiceConnection(
-  "https://xc-mainnet.pyth.network", // <= NOTE: Update this when changing network
-  {
-    priceFeedRequestConfig: {
-      binary: true,
+const parser = yargs(hideBin(process.argv))
+  .scriptName("create_all_price_feeds.ts")
+  .usage("Usage: $0 --contract <contract_id> --private-key <private_key>")
+  .options({
+    contract: {
+      type: "string",
+      demandOption: true,
+      desc: "Contract id to update",
     },
-  }
-);
-// ================================================================
-
-const registry = REGISTRY[network];
-const provider = new JsonRpcProvider(
-  new Connection({ fullnode: registry["RPC_URL"] })
-);
+    "private-key": {
+      type: "string",
+      demandOption: true,
+      desc: "Private key of the transaction signer in hex format without leading 0x",
+    },
+  });
 
 async function main() {
-  if (walletPrivateKey === undefined) {
-    throw new Error("Wallet key unset in environment");
-  }
-  const wallet = new RawSigner(
-    Ed25519Keypair.fromSecretKey(Buffer.from(walletPrivateKey, "hex")),
-    provider
+  const argv = await parser.argv;
+  const contract = DefaultStore.contracts[argv.contract] as SuiContract;
+  const walletPrivateKey = argv["private-key"];
+  const endpoint =
+    contract.chain.getId() === "sui_mainnet"
+      ? "https://xc-mainnet.pyth.network"
+      : "https://xc-testnet.pyth.network";
+  const connection = new PriceServiceConnection(endpoint, {
+    priceFeedRequestConfig: {
+      binary: true,
+    },
+  });
+  const keypair = Ed25519Keypair.fromSecretKey(
+    Buffer.from(walletPrivateKey, "hex")
   );
-  console.log("wallet address: ", wallet.getAddress());
+  console.log("wallet address: ", keypair.getPublicKey().toSuiAddress());
 
   // Fetch all price IDs
-  let { data } = await axios.get(price_feed_id_url);
-  const price_feed_ids = data;
+  const price_feed_ids = await connection.getPriceFeedIds();
   console.log("num price feed ids: ", price_feed_ids.length);
 
   // Create price feeds 20 at a time
   for (let chunk of _.chunk(price_feed_ids, 20)) {
```

Comment:
we need to add some sleep logic here, perhaps 10-20 seconds between each iteration of the for-loop, otherwise this script will hit rate-limits and fail to create all price feeds